tests/System.IO.Pipelines.Tests/PipePoolTests.cs

```diff
@@ -93,11 +93,33 @@ public class PipePoolTests
             await pipe.Writer.WriteAsync(new byte[writeSize]);
         }
 
+        [Fact]
+        public async Task RentsMinimumSegmentSize()
+        {
+            var pool = new DisposeTrackingBufferPool();
+            var writeSize = 512;
+
+            var pipe = new Pipe(new PipeOptions(pool, minimumSegmentSize: 2020));
+
+            var buffer = pipe.Writer.Alloc(writeSize);
+            var allocatedSize = buffer.Buffer.Length;
+            buffer.Advance(buffer.Buffer.Length);
+            buffer.Ensure(1);
+            var ensuredSize = buffer.Buffer.Length;
+            await buffer.FlushAsync();
+
+            pipe.Reader.Complete();
+            pipe.Writer.Complete();
+
+            Assert.Equal(2020, ensuredSize);
```

Comment:
Should these ensure that the size is at least 2020, not that it is exactly 2020?

Reply:
It uses mock implementation of pool that returns exactly what asked.

Reply:
So for 512 request it has to return minimum sized segment.

Reply:
Your call on the PR (i.e. my comment is non blocking), but you don't want to test the mocked pool. You just want to make sure small buffers are not returned. Also, the test could be reused with other pool (that don't return exact size), if you changed the check to >=

Reply:
I'm testing pipe implementation and verifying if it requests the right thing from the pool. It's not specific to pool implementation.